### DISCUSSION TITLE
Is there a way not to wait for `$.ajax` to complete?

### DISCUSSION THREAD

**Discussion by justinjalandoniiwest** [jquery/jquery#4947] [Q&A]
Hello, 

It seems that if multiple `$.ajax` requests are running, 
subsequent requests will not run unless the previous requests finish running.

Is there a way to run `$.ajax` without it having to wait for it to finish, similar 
to fire-and-forget tasks? In essence we would like to make it run similar to a
Thread.

I think the behavior can be done when doing `fetch` requests but not `await`ing.
Is there a way to do this in jQuery?

Thank You

**Comment by mgol** [comment]
There's nothing in jQuery that makes it impossible to run multiple AJAX requests in parallel. If you run too many of them at the same time, browser limits may keep some of them waiting, though. jQuery cannot do anything about these issues.

If you still can't make sense out of it, I'd advise you to post a more complete example on Stack Overflow to get some hints.

**Comment by justinjalandoniiwest** [comment]
> 
> 
> There's nothing in jQuery that makes it impossible to run multiple AJAX requests in parallel. If you run too many of them at the same time, browser limits may keep some of them waiting, though. jQuery cannot do anything about these issues.
> 
> If you still can't make sense out of it, I'd advise you to post a more complete example on Stack Overflow to get some hints.

The browser, I think, queues up the `success` or `error` callback and waits for each to finish in the order the requests were made.

**Comment by mgol** [comment]
> The browser, I think, queues up the `success` or `error` callback and waits for each to finish in the order the requests were made.

That's not true, there's no queuing like that. Responses may arrive at different moments and in a random order and the callbacks will be invoked in that order as well.

**Comment by justinjalandoniiwest** [comment]
> 
> 
> > The browser, I think, queues up the `success` or `error` callback and waits for each to finish in the order the requests were made.
> 
> That's not true, there's no queuing like that. Responses may arrive at different moments and in a random order and the callbacks will be invoked in that order as well.

hello I would like to clarify if older versions of jQuery are also like this? We use jQuery-3.3.1. It really does function that way. I think this can be tried out, try running a long running process on a page, then try to make another request. The second request won't run until the first one finishes

**Comment by mgol** [comment]
Yes, older versions also behave like that. With HTTP/1.1 browsers often limit themselves to 4 parallel connections. If you already have 3 long ones running and you trigger 2 additional `jQuery.ajax` ones then they will run consecutively.

**Comment by justinjalandoniiwest** [comment]
Is there an explanation why it seems to be queuing up the callbacks? We have a long running process and subsequent requests only run once that long running process finishes

**Comment by mgol** [comment]
I cannot reply to that without diving deeper into your app which is beyond the scope of a bug tracker. As I suggested above, try other venues like Stack Overflow.

**Comment by justinjalandoniiwest** [comment]
> 
> 
> I cannot reply to that without diving deeper into your app which is beyond the scope of a bug tracker. As I suggested above, try other venues like Stack Overflow.

i would like to ask if there is a way for $.ajax to not wait for the callback of `success` or `error`?

**Comment by mgol** [comment]
I'm sorry but I don't understand the question.

**Comment by justinjalandoniiwest** [comment]
i would like to clarify. Is there a way to run `$.ajax` but ignore the result so that way we can ignore whether the request finished or had an error?

**Comment by mgol** [comment]
If you don't attach any handlers to your `$.ajax` call then that's exactly what will happen.

**Reply by justinjalandoniiwest** [reply]
> 
> 
> If you don't attach any handlers to your `$.ajax` call then that's exactly what will happen.

Does that mean doing something similar to this:
```js
$.ajax({
    url: requestUrl,
    type: "POST",
    data: formData,
    contentType: false,
    processData: false
});
```

**Reply by aayushs879** [reply]
@mgol @justinjalandoniiwest Is there a way to achieve this for get requests. I have also noticed a similar problem.

**Comment by mgol** [comment]
Yes
-- 
Michał Gołębiowski-Owczarek

**Reply by justinjalandoniiwest** [reply]
thank you, i will try it

**Reply by aayushs879** [reply]
@justinjalandoniiwest did it work?

**Reply by jadejahardipsinh** [reply]
@justinjalandoniiwest - There is no built in limitation in the Jquery but your browser has some limitation to execute maximum parallel connections.

Each browser has a different default value for the maximum parallel connections. For example, in Google Chrome, the default value is 6 connections per domain. This means that Chrome will only make a maximum of 6 HTTP requests to a single domain at the same time. If the website has more than 6 resources (such as images, CSS files, and JavaScript files) that need to be loaded, the browser will queue the additional requests until one of the existing connections becomes available.

The maximum parallel connections can be adjusted in the browser’s settings. However, changing this value can have unintended consequences, such as increasing the load on the server and potentially causing it to crash. It is generally best to leave this setting at the default value unless there is a specific reason to change it.

Reference: https://www.geeksforgeeks.org/what-are-max-parallel-http-connections-in-a-browser/
